### ISSUE TITLE
Cannot use `async` as an object key

### ISSUE THREAD

**Issue by 4141done** [nginx/njs#575]
## Background
A library I'm bundling into my njs script has an options object that is referenced throughout the code that contains the key `async` since the library apparently can work synchronously or not.  The key `async` may often be seen in this kind of code if the code has two modes of operation.

It's difficult to fix this with code substitution since `async` is in use correctly throughout the code. Changing `{ async: true }` to `{ doAsync: true}`  is tough since you have to change references throughout the code, and it's not easy to know whether occurrences of `.async` are referring specifically to the object property you changed.

## Expected
The following code works in both njs and node:
```javascript
const foo = {
  break: 1
};

foo.break;
// 1
```
and in general, any reserved word may be used as an object key when interpreted as an `ObjectExpression` without additional quoting or escaping (see the attached file for some tests I did).

Therefore I would expect that this would work with `async`:

```javascript
const foo = {
  async: 1
};

foo.async;
// 1
```

## Actual
```
09/08/2022 04:48:36 PM > njs
interactive njs 0.7.8

v.<Tab> -> the properties and prototype methods of v.

>> const foo = { async: 1 };
Thrown:
SyntaxError: Unexpected token "async" in shell:1
```

## Workaround
Currently I can get around this by defining the object with the key as a string literal: 
`const foo = { 'async': 1 };`  but this means for third party libraries I need to add code to handle this to my transpilation step.

## Interesting Notes
These are not related to the requested fix but was some interesting learning when trying to understand expected behavior.
This may be very obvious to some but it was new to me so I thought I'd include it:
In both njs and node the file attached called `var_test.js` runs fine (the `async` example must be removed from the njs run for it to work).
```
> node var_test.js 
{ await: 'foo' }
{ break: 'foo' }
...


> njs var_test.js 
{await:'foo'}
{break:'foo'}
...
```

However, when I run it without the variable assignment node and njs both generate an error:
```
> node non_var_test.js 
non_var_test.js:2
{ break: 'foo' };
       ^

SyntaxError: Unexpected token ':'


> njs non_var_test.js 
Thrown:
SyntaxError: Unexpected token ":" in non_var_test.js:2
```

Based on some asking around I did which led to running it through [esprima](https://esprima.org/) it seems that the assignment causes the engine to categorize the statement differently:
```javascript
> console.log(JSON.stringify(esprima.parse('const foo = { bar: 1 };')));
{
  "type": "Program",
  "body": [
    {
      "type": "VariableDeclaration",
      "declarations": [
        {
          "type": "VariableDeclarator",
          "id": {
            "type": "Identifier",
            "name": "foo"
          },
          "init": {
            "type": "ObjectExpression",
            "properties": [
              {
                "type": "Property",
                "key": {
                  "type": "Identifier",
                  "name": "bar"
                },
                "computed": false,
                "value": {
                  "type": "Literal",
                  "value": 1,
                  "raw": "1"
                },
                "kind": "init",
                "method": false,
                "shorthand": false
              }
            ]
          }
        }
      ],
      "kind": "const"
    }
  ],
  "sourceType": "script"
}
> console.log(JSON.stringify(esprima.parse('{ bar: 1 };')));
{
  "type": "Program",
  "body": [
    {
      "type": "BlockStatement",
      "body": [
        {
          "type": "LabeledStatement",
          "label": {
            "type": "Identifier",
            "name": "bar"
          },
          "body": {
            "type": "ExpressionStatement",
            "expression": {
              "type": "Literal",
              "value": 1,
              "raw": "1"
            }
          }
        }
      ]
    },
    {
      "type": "EmptyStatement"
    }
  ],
  "sourceType": "script"
}
console.log(JSON.stringify(esprima.parse('{ bar: 1 }')));
{
  "type": "Program",
  "body": [
    {
      "type": "BlockStatement",
      "body": [
        {
          "type": "LabeledStatement",
          "label": {
            "type": "Identifier",
            "name": "bar"
          },
          "body": {
            "type": "ExpressionStatement",
            "expression": {
              "type": "Literal",
              "value": 1,
              "raw": "1"
            }
          }
        }
      ]
    }
  ],
  "sourceType": "script"
}
```

So in the `ObjectExpression` context the node parser is ok with reserved words as keys, but in the `BlockStatement` context it is not.

Another interesting thing is that in node the statement `{ break: 1}` (no `;`) is accepted, whereas in njs it fails with the same error.  I suspect this must be something to do with automatic semicolon insertion but I am not sure why.

**Comment by 4141done** [comment]
Forgot that I cannot attach files, so here's a gist with some of the tests

https://gist.github.com/4141done/5a196206eace6c19ff54985b2de921bd

**Comment by xeioex** [comment]
Hi @4141done,

Thanks for the report, will take a look next week.

**Comment by xeioex** [comment]
@4141done 

Funny thing:

```
cat t.js
{ break: 1}

node t.js
{ break: 1}
       ^

SyntaxError: Unexpected token ':'
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1025:15)
    at Module._compile (node:internal/modules/cjs/loader:1059:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1124:10)
    at Module.load (node:internal/modules/cjs/loader:975:32)
    at Function.Module._load (node:internal/modules/cjs/loader:816:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:79:12)
    at node:internal/main/run_main_module:17:47
    
njs t.js
Thrown:
SyntaxError: Unexpected token ":" in t.js:1
```

**Comment by VadimZhestikov** [comment]
In this case "{break: 1}" is parsed as block statement. In strict mode reserved word "break" can't be used as label. It is ok.
But node 16.4.0 wrongly reject this script in non strict mode. In 16.16.0 it is already fixed (while in strict mode node 16.16.0 reject it as expected).

**Comment by 4141done** [comment]
Apologies for the delay in my response.  I've tested this out on my end and things are working perfectly.  As usual I really appreciate your attention and once again learned something new from this one 🥇
